spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jms/activemq/ActiveMQConnectionFactoryFactory.java

```diff
@@ -59,6 +61,9 @@
 	private <T extends ActiveMQConnectionFactory> T doCreateConnectionFactory(
 			Class<T> factoryClass) throws Exception {
 		T factory = createConnectionFactoryInstance(factoryClass);
+		MutablePropertyValues propertyValues =
+				new MutablePropertyValues(this.properties.getProperties());
+		new RelaxedDataBinder(factory).bind(propertyValues);
```

Comment:
That is not what I had in mind. Your report is about a `Properties` object that you set  in the factory via `setProperties` and the reason why this issue exists is that because binding doesn't do its job. 

What am I missing? Can't you just call `setProperties` with the properties object that you have there? It's not the same thing as binding on actual java bean properties (and, in effect, any side effect that could happen due to the automatic binding _will_ occur here as well since that same code runs effectively with `@ConfigurationProperties`).